src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt

```diff
@@ -233,6 +239,48 @@ private fun Int.fromBodyRelativeOffset(call: RsMacroCall): Int? {
     return elementOffset
 }
 
+private fun MappedTextRange.fromBodyRelativeRange(call: RsMacroCall): MappedTextRange? {
+    val newSrcOffset = srcOffset.fromBodyRelativeOffset(call) ?: return null
+    return MappedTextRange(newSrcOffset, dstOffset, length)
+}
+
+fun RsMacroCall.mapRangeFromExpansionToCallBodyStrict(range: TextRange): TextRange? {
```

Comment:
As we discussed in person, let's convert it from extension function into a common one with `RsMacroCall` as the first argument. Looks like it has more sense
To avoid another one weird function in public scope, let's create util class and move all functions related to expansion and range mapping into it
Don't insist to do it in this PR